tests/sycl/test_barrier.sycl

```diff
@@ -0,0 +1,53 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <CL/sycl.hpp>
+
+#include "../../device/sycl/src/utils/barrier.hpp"
+
+TEST(CUDABarrier, BarrierAnd) {
+    ::sycl::queue queue;
+    constexpr std::size_t n_bools = 4;
+
+    bool *out = malloc_shared<bool>(n_bools, queue);
+
+    queue.submit([&](cl::sycl::handler &h) {
+        h.parallel_for(cl::sycl::range<1>(128), [=](::sycl::nd_item<1> item) {
+            traccc::sycl::barrier bar(item);
+
+            bool v;
+
+            v = bar.blockAnd(false);
+            if (threadIdx.x == 0) {
+                out[0] = v;
+            }
+
+            v = bar.blockAnd(true);
+            if (threadIdx.x == 0) {
+                out[1] = v;
+            }
+
+            v = bar.blockAnd(threadIdx.x % 2 == 0);
+            if (threadIdx.x == 0) {
+                out[2] = v;
+            }
+
+            v = bar.blockAnd(threadIdx.x < 32);
+            if (threadIdx.x == 0) {
+                out[3] = v;
+            }
+        });
+    });
+
+    ASSERT_FALSE(out[0]);
+    ASSERT_TRUE(out[1]);
+    ASSERT_FALSE(out[2]);
+    ASSERT_FALSE(out[3]);
```

Comment:
```suggestion
    EXPECT_FALSE(out[0]);
    EXPECT_TRUE(out[1]);
    EXPECT_FALSE(out[2]);
    EXPECT_FALSE(out[3]);
```

So that all failures would be reported by the test.